patch_recurrence/steps.py

```diff
@@ -39,10 +40,53 @@ def generate_patches(scaled_imgs, constants):
         patches.append(img_patches)
     return patches
 
+#..............................................................................................#
+# purva's functions
+
+# exactly like generate_patches without the std deviation condition
+def generate_all_patches(scaled_imgs, constants):
+    patch_size = constants.PATCH_SIZE
+    patches = []
+    for k, sc in enumerate(scaled_imgs):
+        img_patches = []
+        for i in range(sc.shape[0] - patch_size):
+            for j in range(sc.shape[1] - patch_size):
+                raw_patch = sc[i:i + patch_size, j:j + patch_size, :]
+                patch = Patch(
+                    raw_patch=raw_patch,
+                    patch_size=patch_size,
+                )
+                patch.store(sc, [i, j])
+                img_patches.append(patch)
+        patches.append(img_patches)
+    return patches
+
+
+# assigning bucket numbers to each patch
+def set_patch_buckets(patches, constants):
+    num_buckets = constants.NUM_BUCKETS
+    scaled_imgs = len(patches)
+    std_database = np.zeros((len(patches), len(patches[0])))
+    for k in range(scaled_imgs):
+        for index, patch in enumerate(patches[k]):
+            std_database[k][index] = patch.std_dev
```

Comment:
This can be done in an easier way.
````
std_database = np.zeros((height, width))
for patch in img:
    std_database[patch.location] = patch.std_dev
````
There is a bit of an offset (check how patches are extracted), so you might need to worry about that